packages/moments/src/client/getMomentsDefaultClient.ts

```diff
@@ -0,0 +1,73 @@
+import { MomentsClient } from 'moments';
+import {
+  AuthenticationProvider,
+  AuthenticationProviderHttp,
+  PoapCompass,
+  PoapMomentsApi,
+} from '@poap-xyz/providers';
+
+interface GetMomentsDefaultClientOptions {
+  moments: {
+    momentsBaseUrl?: string;
+    /**
+     * The authentication provider used for authenticating requests to the moments API.
+     * If this is provided, oAuthAuthentication will be ignored.
+     */
+    authenticationProvider?: AuthenticationProvider;
+    /**
+     * The OAuth authentication configuration used for authenticating requests to the moments API.
+     * If this is provided, it will be used to create an `AuthenticationProviderOAuth` instance.
+     */
+    oAuthCredentials?: {
+      clientId: string;
+      clientSecret: string;
+      oAuthServerDomain?: string;
+    };
+  };
```

Comment:
You can do union to enforce one or the other with types.

Something like:

```
  moments: { momentsBaseUrl?: string; } & (
    | { authenticationProvider: ... }
    | { oAuthCredentials: { ... }
  );
```

Reply:
Thanks!